df_chat/tests/test_message_endpoint.py

```diff
@@ -27,6 +27,42 @@ def test_message_creation_success(self):
         self.assertIsNone(message["parent_id"])
         self.assertFalse(message["is_reaction"])
 
+    def test_message_creation_failure_unauthenticated(self):
+        """
+        Testing failure to create a message when the user is not authenticated.
+        """
+        room = self.create_room_and_add_users()
+
+        message_endpoint = reverse("rooms-messages-list", kwargs={"room_pk": room.pk})
+        response = self.client.post(message_endpoint, {"body": "Hi"})
+        self.assertEqual(response.status_code, 401)
+
+    def test_message_creation_failure_user_not_in_room(self):
+        """
+        Testing failure to create a message when the user is not part of a room.
+        """
+        user, token = self.create_user()
+        room = self.create_room_and_add_users()
+
+        message_endpoint = reverse("rooms-messages-list", kwargs={"room_pk": room.pk})
+        self.client.credentials(HTTP_AUTHORIZATION="Bearer " + token)
+        response = self.client.post(message_endpoint, {"body": "Hi"})
+        self.assertEqual(response.status_code, 201)
```

Comment:
This looks like an issue to me. If a user is not part of a room and he is trying to create a message and fails, then why is the status a 201 success?

IIUC the message got created. It shouldn't.

Reply:
Sorry about that! Late night coding sure takes a toll. I've revised it, this is because the room is automatically set to be Public which is why the message was still created. There are 2 approaches that could be done. In the `RoomFactory` I can directly just set `is_public` to False but I think that is counterintuitive because the project itself has a default value of `True` for is_public. Instead I have created a parameter for the `create_room_and_add_users` function to take in an extra variable of `ispublic` I did not use camel case because of python's lint. 
If you think this is a good and acceptable approach, please feel free to resolve the comment, thank you!